### ISSUE TITLE
use a reliable service/SCM for your repository

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#576]
```
What steps will reproduce the problem?
1. try to clone
2. try to pull
3. wait

nothing happens, it's stuck at "requesting all changes".

Sometimes you wait for minutes, then it works. Sometimes not.
That alienates people who want to contribute or just want to use the latest 
changeset and is bad for the project as a whole.

I have no idea if this is related to google or to mercurial.
It is not my internet connection. I tested with several other repositories, 
including ogre mercurial from bitbucket (not sure if the history is comparable 
though).
Or just try cloning scummvm from github which has an even bigger history than 
hedgewars.
```

Original issue reported on code.google.com by `julian.o...@googlemail.com` on 15 Mar 2013 at 8:57

**Comment by GoogleCodeExporter** [comment]
```
Look. All services have problems sometimes.  This isn't helpful.

```

Original comment by `kyberneticist@gmail.com` on 16 Mar 2013 at 12:54
* Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
I suggest, if you think it is a problem w/ mercurial to file a bug w/ *them* 
and suggest they try cloning this repo.
I also suggest, if you think this is a bug w/ google code, to file a bug w/ 
*them* as well...

```

Original comment by `kyberneticist@gmail.com` on 16 Mar 2013 at 12:55

**Comment by GoogleCodeExporter** [comment]
```
how about mirroring the repository somewhere else which will enable testing 
this case and eventually make more people contribute
```

Original comment by `julian.o...@googlemail.com` on 16 Mar 2013 at 12:58

**Comment by GoogleCodeExporter** [comment]
```
Amusingly, we moved to Google Code because former host was down too often.  If 
others were consistently reporting what you were saying, would be more inclined 
to take this seriously, and consider the *SIGNIFICANT* hassle of moving 
everything to bitbucket (all our links referencing google code, the wiki, 
people w/ who'd come to rely on us being here...)

As it is, yeah.  I've *occasionally* had a pull hang, which almost always 
resolves itself as soon as I abort and pull again, which makes me wonder if 
they just have a bad machine in some cluster or somesuch.

But it doesn't happen very often, and not to point of going to that much 
trouble.

So, I really do suggest a bug.  Against Google Code is probably best bet.

They fixed the last one I reported.

```

Original comment by `kyberneticist@gmail.com` on 16 Mar 2013 at 1:02

**Comment by GoogleCodeExporter** [comment]
```
I very much doubt this is having any significant impact on contribution BTW ☺

```

Original comment by `kyberneticist@gmail.com` on 16 Mar 2013 at 1:03

**Comment by GoogleCodeExporter** [comment]
```
I'd just like to add that if you try from non google server and you are not 
sure about the history size is comparable, you're doing a pointless test.
```

Original comment by `vittorio...@gmail.com` on 16 Mar 2013 at 1:04

**Comment by GoogleCodeExporter** [comment]
```
Well, I don't do testing on latest head anymore because of this annoyance.

And as said... the scummvm history is bigger, so yes, it is comparable. In both 
cases mercurial was slower.
```

Original comment by `julian.o...@googlemail.com` on 16 Mar 2013 at 1:20

**Comment by GoogleCodeExporter** [comment]
```
Cloned over http?
Also, hedgewars history might be more complex...

Actually, reason ages ago git was inappropriate was 'cause it was horribly slow 
over http compared to mercurial.
They finally fixed that a year or two ago, but it isn't like it is *faster* ☺

But anyway...
You're a bit off on scummvm...

$ time git clone git://github.com/scummvm/scummvm.git
Cloning into 'scummvm'...
remote: Counting objects: 439668, done.
remote: Compressing objects: 100% (74866/74866), done.
remote: Total 439668 (delta 365518), reused 437252 (delta 363641)
Receiving objects: 100% (439668/439668), 114.71 MiB | 1.87 MiB/s, done.
Resolving deltas: 100% (365518/365518), done.

real    1m29.543s
user    0m29.530s
sys 0m3.740s

$ du -hs scummvm/.git
128M    scummvm/.git
(1.42MiB/s)

$ time hg clone -v https://hedgewars.googlecode.com/hg/ hw
[snip]
5895 files updated, 0 files merged, 0 files removed, 0 files unresolved         

real    5m34.822s
user    1m23.273s
sys 0m8.609s

$ du -hs hw/.hg
595M    hw/.hg
(1.78 MiB/s)

Hedgewars is almost 5x the size of scummvm. So. Google Code cloned hedgewars 
25% faster.
Hedgewars also has 3x the files, but that really doesn't impact cloning, since 
that's just handled in final unpack and is a negligible part of the total

```

Original comment by `kyberneticist@gmail.com` on 16 Mar 2013 at 2:36

**Comment by GoogleCodeExporter** [comment]
```
I was talking about the history, not the repository size. And for me it's even 
5 minutes for _pulling_, so that seems to me like a mercurial quirk when 
dealing with large histories.

Repository size depends on more factors than just history.
```

Original comment by `julian.o...@googlemail.com` on 16 Mar 2013 at 2:46

**Comment by GoogleCodeExporter** [comment]
```
And, testing a quick update.
$ hg strip b08cc68a26a3
15 files updated, 0 files merged, 0 files removed, 0 files unresolved
saved backup bundle to /tmp/hw/.hg/strip-backup/b08cc68a26a3-backup.hg
$ time hg pull -u
pulling from https://hedgewars.googlecode.com/hg/

searching for changes
adding changesets
adding manifests
adding file changes
added 10 changesets with 16 changes to 13 files (-1 heads)
15 files updated, 0 files merged, 0 files removed, 0 files unresolved

real    0m2.696s
user    0m0.352s
sys 0m0.044s

So, pulling a set of 10 changes happened plenty fast.

For a larger strip...
added 102 changesets with 290 changes to 99 files (-2 heads)
102 files updated, 0 files merged, 3 files removed, 0 files unresolved

real    0m14.816s
user    0m0.632s
sys 0m0.088s

Slightly longer on the "searching for changes" set, but still nbd.


In short. Not worth the effort, google code mercurial is performing quite well. 
Bug stays closed, but if you're having any trouble connecting, I strongly 
suggest reporting to google code.

```

Original comment by `kyberneticist@gmail.com` on 16 Mar 2013 at 2:46

**Comment by GoogleCodeExporter** [comment]
```
Anyway, it seems to be faster today than yesterday, so maybe it's a google 
problem after all.

Github does not seem to be better at the moment and has massive 404 problems.
```

Original comment by `julian.o...@googlemail.com` on 16 Mar 2013 at 2:53
